dashboard/src/api/auth-manager.ts

```diff
@@ -0,0 +1,138 @@
+import * as axios from 'axios';
+
+import { Api } from './index';
+import { AuthToken, CodeToken } from './types';
+
+class TokenStorage {
+    storeToken(token: string) {
+        if (!token) {
+            throw new Error('Cannot store empty token');
+        }
+        localStorage.setItem('AUTH_TOKEN', token);
+        localStorage.setItem('AUTH_TOKEN_STORED_AT', Date.now().toString());
+    }
+
+    getToken(): string | null {
+        return localStorage.getItem('AUTH_TOKEN');
+    }
+
+    clearToken() {
+        localStorage.removeItem('AUTH_TOKEN');
+        localStorage.removeItem('AUTH_TOKEN_STORED_AT');
+    }
+
+    getTokenAge(): number | null {
+        const timestampStr = localStorage.getItem('AUTH_TOKEN_STORED_AT');
+        if (!this.getToken() || !timestampStr) {
+            return null;
+        }
+        return Date.now() - Number(timestampStr);
+    }
+}
+
+const tokenStorage = new TokenStorage();
+
+export default class AuthManager {
+    /** Maximum age of the token in seconds before it is refreshed. */
+    public maxTokenAge: number = 5 * 60 * 1000;  // 5 minutes
+
+    private _api: Api;
```

Comment:
Style bikeshedding suggestion:
In JS it is usually regarded as bad to add underscore to variable names, and MS TypeScript guide seems to agree regarding TS [1] as well. But it seems to be a bikeshedding issue, for instance the Angular team has discussed this as well[2]. No need to change I guess, just wanted to point it out.

[1] https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names
[2] https://github.com/angular/angular.io/issues/1108

Reply:
So, what's your suggestion for private variable naming?

Reply:
`private api: Api;` ?